common/lib/xmodule/xmodule/capa_base.py

```diff
@@ -659,7 +659,7 @@ def get_problem_html(self, encapsulate=True):
             check_button_checking = False
 
         content = {
-            'name': self.display_name_with_default_escaped,
+            'name': self.display_name.strip() or self.location.block_type,
```

Comment:
I wonder if we should instead override the display_name_with_default method.

Also, I agree that it is correct to remove the "escaped" part, but we have to make sure that all HTML that is using this is correctly escaping.